### PR TITLE
fix: help and version functionality

### LINKED ISSUE #1917: Inconsistent behavior of webpack --flag --help
**Describe the bug**

`webpack --flag --help`  shows inconsistent behavior, sometimes it prints help sometimes it considers `--help` as value for `--flag`.

**What is the current behavior?**

**To Reproduce**

Try running -

`webpack --stats --help`
`webpack --mode --help`
`webpack -hot --help`
`webpack --entry --help`

See inconsistency.

**Expected behavior**
The behavior should be consistent either print help (recommended) or throw an error/warning.

**Screenshots**
Screenshot at 2020-10-11 08-11-21

**Please paste the results of `webpack-cli info` here, and mention other relevant information**

**Additional context**
https://github.com/webpack/webpack.js.org/pull/4051#discussion_r502856883

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
bugfix
Fixes #1917 

**Did you add tests for your changes?**
Yes
**If relevant, did you update the documentation?**
NA
**Summary**

Screenshot at 2020-10-20 15-59-27

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

File: packages/webpack-cli/lib/utils/arg-parser.js [hunk 1]
```diff
@@ -38,11 +38,11 @@ const argParser = (options, args, argsOnly = false, name = '') => {
     // Prevent default behavior
     parser.on('command:*', () => {});
 
-    // Use customized help output if available
-    parser.on('option:help', () => {
+    // Use customized help output
+    if (args.includes('--help')) {
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/utils/arg-parser.js:42]: Do wee need https://github.com/webpack/webpack-cli/blob/master/packages/webpack-cli/lib/bootstrap.js#L25?

Comment (snitin315) [reply packages/webpack-cli/lib/utils/arg-parser.js:42]: Good catch. It should be removed from there.

Comment (snitin315) [reply packages/webpack-cli/lib/utils/arg-parser.js:42]: The same should be done for `version` too.

Screenshot at 2020-10-20 16-20-20

Comment (snitin315) [reply packages/webpack-cli/lib/utils/arg-parser.js:42]: I will update 👍🏻

**Review by evilebottnawi — APPROVED** [review]
/cc @webpack/cli-team

File: test/help/help-commands.test.js [hunk 2]
```diff
@@ -4,11 +4,6 @@ const { run } = require('../utils/test-utils');
 const helpHeader = 'The build tool for modern web applications';
 
 describe('commands help', () => {
-    it('throws error if supplied as an argument for subcommands', () => {
-        const { stderr } = run(__dirname, ['serve', 'help'], false);
```
Comment (jamesgeorge007) [inline test/help/help-commands.test.js:8]: Supplying help as an argument to any of the subcommands shouldn't be allowed.

Comment (snitin315) [reply test/help/help-commands.test.js:8]: I think we can allow it for consistency.

`webpack --flag help` - Allowed
`webpack help --help` - Allowed
`webpack  help cmd` - Allowed

Comment (evilebottnawi) [reply test/help/help-commands.test.js:8]: `webpack help --help` interesting, but I think it is good

Comment (snitin315) [reply test/help/help-commands.test.js:8]: Ah, I wanted to type `webpack help --flag`.
Just checked `webpack help --help` is allowed as well and is equivalent to `webpack help`

Comment (evilebottnawi) [reply test/help/help-commands.test.js:8]: /cc @jamesgeorge007

File: packages/webpack-cli/lib/bootstrap.js [hunk 3]
```diff
@@ -11,18 +9,8 @@ process.title = 'webpack-cli';
...
-    }
-
     const commandIsUsed = isCommandUsed(cliArgs);
-
-    if (parsedArgs.unknownArgs.includes('version') || parsedArgs.opts.version) {
-        versionRunner(cliArgs, commandIsUsed);
-        process.exit(0);
-    }
-
+    const parsedArgs = parseArgs(cliArgs);
```
Comment (snitin315) [inline packages/webpack-cli/lib/bootstrap.js:13]:
```suggestion
```

File: test/help/help-commands.test.js (hunk 2 above)
Comment (evilebottnawi) [reply test/help/help-commands.test.js:8]: @snitin315 Can we refactor this test to allow using `help` here

Comment (snitin315) [reply test/help/help-commands.test.js:8]: Yes.

Comment (snitin315) [reply test/help/help-commands.test.js:8]: updated 👍🏻

**Review by evilebottnawi — APPROVED** [review]
/cc @webpack/cli-team need review

**Review by jamesgeorge007 — APPROVED** [review]

**Pull request merged**